### ISSUE TITLE
 list outputs doesn't return error if credentails are missing

### ISSUE THREAD

**Issue by m1keil** [Sceptre/sceptre#609]
When doing action such as deploy without credentials loaded in environment, and error message will be displayed: `Partial credentials found in assume-role, missing: source_profile or credential_source`

However, when doing list outputs, the command will silently return an empty list which my trick you to think there's no outputs instead of the list didn't actually work.

**Comment by ngfgrant** [comment]
Yeh good spot I think I noticed this too before. Probably worth being consistent and throwing an error for list outputs too!

**Comment by ngfgrant** [comment]
I think it would be better to keep the try/except but we could potentially handle the except better.

**Comment by m1keil** [comment]
What do you have in mind? I believe currently in most places around Sceptre currently credentials exceptions are just left to go all the way up.

Maybe best way is to catch credentials related exceptions all the way at the top somewhere in the CLI, instead of handling it on each action individually?

**Comment by ngfgrant** [comment]
I'd prefer to catch as soon as we can within the core API rather than CLI so that those using sceptre as a module will benefit too. 

Probably makes sense to add the check in somewhere in `connection_manager.py` (https://github.com/cloudreach/sceptre/blob/master/sceptre/connection_manager.py). Just a quick look at that file I can see a couple of areas where we can improve things. Maybe in the instantiation of connection_manager if profile is None raise a NoCredentialsSupplied error?

**Comment by m1keil** [comment]
aight sounds like a good idea. I'll check the code and see if I can make a PR.

**Comment by m1keil** [comment]
Sorry for the long period of no activity here, been a bit busy IRL.
I've been reviewing this today, here some of my notes:

Permission related error scenarios:
1. No `AWS_*` envars or profile supplied.
    `botocore.exceptions.NoCredentialsError: Unable to locate credentials` is raised. 

2. Profile doesn't exist in `~/.aws/config` (no envars provided).
    `botocore.exceptions.ProfileNotFound: The config profile (<profile_name>) could not be found` is raised.

3. Invalid credentials.
    `botocore.exceptions.ClientError: An error occurred (InvalidClientTokenId) when calling the <OP> operation: The security token included in the request is invalid.` is raised.

4. Expired credentials.
    `botocore.exceptions.ClientError: An error occurred (ExpiredToken) when calling the <OP> operation: The security token included in the request is expired`
    This will be common if you using dynamic short lived credentials from STS and not static ones.

5. Partial profile.
    `botocore.exceptions.PartialCredentialsError: Partial credentials found in assume-role, missing: source_profile or credential_source`
    This is the original problem reported. This will most likely happen if you have a profile configued with `role_arn` but no credentials provided to assume that role (either envars or in config files).

I think that all these three scenarios fall under the "unexceptional exceptions" scenario, which probably should be handled by Sceptre. The user should know if the action failed because of permissions problem.

Scenario 3 and 4 is what cause commands like `sceptre list outputs stack` to return `[]` instead of credentials error because of the very generous try/catch for `botocore.exceptions.ClientError`.

This becomes more tricky by the fact that these exceptions are being thrown at different stages. Some will be raised when creating sessions, others when creating client and the rest when making the actual API call.

So I guess what we can do, is to catch these in `connection_manager.py` and wrap with Sceptre's own exception to allow easier handling down the line?

P.S 
Regarding your initial suggestion:
> just a quick look at that file I can see a couple of areas where we can improve things. Maybe in the instantiation of connection_manager if profile is None raise a NoCredentialsSupplied error
I believe this is not possible as the `profile` config is not mandatory and sceptre can run with just envars provided?

**Comment by m1keil** [comment]
@ngfgrant  Submitted initial PR. Let me know if you think that's the way to go.

**Comment by m1keil** [comment]
This will also fix #621

**Comment by ngfgrant** [comment]
@m1keil could you checkout 2.1.4 branch to see if this has been resolved?

**Comment by zaro0508** [comment]
I believe this is fixed in sceptre version 2.6

```
➜ sceptre --version
Sceptre, version 2.6.3

➜ sceptre list outputs prod/bootstrap.yaml
"Session credentials were not found. Profile: None. Region: us-east-1."

➜ mv ~/.aws/config ~/.aws/config-bak
➜ sceptre list outputs prod/bootstrap.yaml
"Session credentials were not found. Profile: None. Region: us-east-1."

➜ AWS_PROFILE=itsandbox-admin sceptre list outputs prod/bootstrap.yaml
"The config profile (itsandbox-admin) could not be found"
```
